src/NATS.Client.Core/Internal/ISocketConnection.cs

```diff
@@ -1,6 +1,6 @@
 namespace NATS.Client.Core.Internal;
 
-internal interface ISocketConnection : IAsyncDisposable
+public interface ISocketConnection : IAsyncDisposable
```

Comment:
if we need to make it public we should move the file out of `internal`

Reply:
Yes, if we want to use it in `public Func<ISocketConnection, ValueTask<ISocketConnection>>`. Will move it.